cmd/allocator/main.go

```diff
@@ -193,18 +237,32 @@ func (h *httpHandler) allocateHandler(w http.ResponseWriter, r *http.Request) {
 	logger.WithField("request", request).Infof("allocation request received")
 
 	gsa := converters.ConvertAllocationRequestV1Alpha1ToGSAV1(&request)
-	allocation := h.agonesClient.AllocationV1().GameServerAllocations(gsa.ObjectMeta.Namespace)
-	allocatedGsa, err := allocation.Create(gsa)
+	resultObj, err := h.allocationCallback(gsa, h.stop)
```

Comment:
Why is there a callback defined instead of just calling `h.allocator.Allocate(gsa, stop)` here?

Reply:
It looks like it's necessary for testing.